metagraph/src/annotation/binary_matrix/row_diff/row_diff.cpp

```diff
@@ -1,119 +1,31 @@
 #include "row_diff.hpp"
 
-#include "graph/annotated_dbg.hpp"
+#include "annotation/binary_matrix/column_sparse/column_major.hpp"
 
 namespace mtg {
 namespace annot {
 namespace binmat {
 
-
-bool RowDiff::get(Row row, Column column) const {
-    SetBitPositions set_bits = get_row(row);
-    SetBitPositions::iterator v = std::lower_bound(set_bits.begin(), set_bits.end(), column);
-    return v != set_bits.end() && *v == column;
-}
-
-std::vector<BinaryMatrix::Row> RowDiff::get_column(Column column) const {
-    std::vector<Row> result;
-    for (Row row = 0; row < num_rows(); ++row) {
-        if (get(row, column))
-            result.push_back(row);
-    }
-    return result;
-}
-
-BinaryMatrix::SetBitPositions RowDiff::get_row(Row row) const {
-    Vector<uint64_t> result = get_diff(row);
-
-    uint64_t boss_edge = graph_->kmer_to_boss_index(
-            graph::AnnotatedSequenceGraph::anno_to_graph_index(row));
-    const graph::boss::BOSS &boss = graph_->get_boss();
-
-    while (!terminal()[row]) {
-        graph::boss::BOSS::TAlphabet w = boss.get_W(boss_edge);
-        assert(boss_edge > 1 && w != 0);
-
-        // fwd always selects the last outgoing edge for a given node
-        boss_edge = boss.fwd(boss_edge, w % boss.alph_size);
-        row = graph::AnnotatedSequenceGraph::graph_to_anno_index(
-                graph_->boss_to_kmer_index(boss_edge));
-        merge(&result, get_diff(row));
-    };
-    return result;
-}
-
-bool RowDiff::load(std::istream &f)  {
-    f.read(reinterpret_cast<char *>(&num_columns_), 8);
-    f.read(reinterpret_cast<char *>(&num_relations_), 8);
-    diffs_.load(f);
-    boundary_.load(f);
-    terminal_.load(f);
-    sdsl::util::init_support(sboundary_, &boundary_);
-    return true;
-}
-
-void RowDiff::serialize(std::ostream &f) const  {
-    f.write(reinterpret_cast<const char *>(&num_columns_), 8);
-    f.write(reinterpret_cast<const char *>(&num_relations_), 8);
-    diffs_.serialize(f);
-    boundary_.serialize(f);
-    terminal_.serialize(f);
-}
-
-void RowDiff::serialize(const std::string &name) const {
+template <class BaseMatrix>
+void RowDiff<BaseMatrix>::serialize(const std::string &name) const {
```

Comment:
hm, so it does work. Why not move other methods to .cpp as well?

Reply:
yeah, I see the problem. Though `override` is optional so maybe we could do without specifying it. The code will be easier to read and maybe it will compile faster too. I think it's worth neglecting `override` in this case, and I'm not a huge fan of it anyway.